agent/gcp_labels.go

```diff
@@ -0,0 +1,46 @@
+package agent
+
+import (
+	"context"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+)
+
+type GCPLabels struct {
+}
+
+func (e GCPLabels) Get() (map[string]string, error) {
```

Comment:
Given there aren't any instance variables in the struct, I'd tend to just make this a function, perhaps `GetGCPLabels(ctx context.Context) (map[string]string, error)`?

Reply:
I thought the syntax was a little weird, but it is my first time working with Go so I followed the pattern established in `ec2_tags.go`: https://github.com/buildkite/agent/blob/d6dfcf8ded85579ca40f731c5479cb7c471cd7d0/agent/ec2_tags.go#L9 which is used in `agent_pool.go`: https://github.com/buildkite/agent/blob/d6dfcf8ded85579ca40f731c5479cb7c471cd7d0/agent/agent_pool.go#L214

Happy to change this to a function, but I think the EC2 tags and GCP labels should follow the same pattern. I can open a second PR to switch the EC2 tags to a function too if you'd like.